### PR TITLE
fix(attrs): check key on setAttr

### LINKED ISSUE #11040: v-bind with empty key
### Version
2.6.11

### Reproduction link
[https://jsfiddle.net/andrewharvey4/k6r3uzby/10/](https://jsfiddle.net/andrewharvey4/k6r3uzby/10/)

### Steps to reproduce
v-bind an object with a empty string key

### What is expected?
vue not to crash, it to either be silently ignored or a warning but not a fatal error

### What is actually happening?
an error is thown crashing the app

### PR DESCRIPTION
**Pull request by mateonunez**
verify if key is invalid then catch error

fix #11040

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch for v2.x (or to a previous version branch), _not_ the `master` branch
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] All tests are passing: https://github.com/vuejs/vue/blob/dev/.github/CONTRIBUTING.md#development-setup
- [ ] New/updated tests are included

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

File: src/platforms/web/runtime/modules/attrs.js [hunk 1]
```diff
@@ -18,7 +19,7 @@ import {
   convertEnumeratedValue
 } from 'web/util/index'
 
-function updateAttrs (oldVnode: VNodeWithData, vnode: VNodeWithData) {
+function updateAttrs(oldVnode: VNodeWithData, vnode: VNodeWithData) {
```
Comment (posva) [inline src/platforms/web/runtime/modules/attrs.js]: Can you revert style changes please

File: src/platforms/web/runtime/modules/attrs.js [hunk 2]
```diff
@@ -59,7 +60,8 @@ function updateAttrs (oldVnode: VNodeWithData, vnode: VNodeWithData) {
   }
 }
 
-function setAttr (el: Element, key: string, value: any) {
+function setAttr(el: Element, key: string, value: any) {
```
Comment (posva) [inline src/platforms/web/runtime/modules/attrs.js]: here as well

File: src/platforms/web/runtime/modules/attrs.js [hunk 3]
```diff
@@ -83,12 +85,16 @@ function setAttr (el: Element, key: string, value: any) {
     } else {
       el.setAttributeNS(xlinkNS, key, value)
     }
+  } else if (isEmpty(key)) {
```
Comment (posva) [inline src/platforms/web/runtime/modules/attrs.js]: this is a dev warning, it should not throw but warn **in dev only** like other warnings

File: src/platforms/web/util/attrs.js [hunk 4]
```diff
@@ -45,6 +45,10 @@ export const isXlink = (name: string): boolean => {
   return name.charAt(5) === ':' && name.slice(0, 5) === 'xlink'
 }
 
+export const isEmpty = (name: string): boolean => {
```
Comment (posva) [inline src/platforms/web/util/attrs.js]: This is unnecessary, we can do the check directly on the other file

**Comment by mateonunez** [comment]
Thanks for the notes

**Comment by mateonunez** [comment]
Thanks for the notes

**Comment by yangguansen** [comment]
> Thanks for the commit! I don't understand how this fixes the issue though? Doesn't this just suppress the warning message?

looks like yes

**Pull request closed without merging**